src/pygrep_hooks/plugins.rs

```diff
@@ -0,0 +1,54 @@
+use rustpython_ast::{Expr, ExprKind};
+
+use crate::ast::types::{BindingKind, Range};
+use crate::check_ast::Checker;
+use crate::checks::{Check, CheckKind};
+
+/// PGH002 - deprecated use of logging.warn
+pub fn deprecated_log_warn(checker: &mut Checker, func: &Expr) {
+    match &func.node {
+        ExprKind::Attribute { attr, value, .. } => {
+            let ExprKind::Name { id, .. } = &value.node else { return; };
+
+            if attr != "warn" {
+                return;
+            }
+
+            // `warn.log` is a special case handled by pygrep-hooks
+            if !(attr == "warn" && id == "log") {
+                if !checker
+                    .current_scopes()
+                    .find_map(|scope| scope.values.get(id.as_str()))
+                    .map_or(false, |index| matches!(
+                        &checker.bindings[*index].kind,
+                        BindingKind::Importation(name, full_name) if full_name == "logging" && name == id)
+                    )
+                {
+                    return;
+                }
+            }
+        }
+        ExprKind::Name { id, .. } => {
+            if id != "warn" {
+                return;
+            }
+
+            if !checker
+                .current_scopes()
+                .find_map(|scope| scope.values.get(id.as_str()))
+                .map_or(false, |index| matches!(
+                    &checker.bindings[*index].kind,
+                    BindingKind::FromImportation(name, full_name) if name == "warn" && full_name == "logging.warn")
+                )
+            {
+                return;
+            }
+        }
+        _ => return,
+    }
+
+    checker.add_check(Check::new(
+        CheckKind::DeprecatedLogWarn,
+        Range::from_located(func),
+    ));
```

Comment:
I think I'd find it easier to reason about the logic if we inverted all the cases -- so, instead of early-return, we moved this `add_check` into the few paths that _did_ trigger the error (and made the cases, like, `if id == "warn"`, and so on).

Reply:
I was wondering about that in the beginning as well. Then I just made it work first but didn't reevaluate. Happy to change that.